src/pkcs15init/pkcs15-oberthur-awp.c

```diff
@@ -996,6 +996,20 @@ awp_encode_cert_info(struct sc_pkcs15_card *p15card, struct sc_pkcs15_object *ob
 	/*
 	 * serial number
 	 */
+#if 1
+	
+	/* TODO the der encoding of a ANS1_INTEGER is a TLV, the original code only as using the 
+	 * i2c_ASN1_INTEGER which is not in OpenSSL 1.1  
```

Comment:
Sorry I have no way to test this and i2c_ASN1_INTEGER is  not
    supported in OpenSSL-1.1, so some change has to be made. 
    Someone who has the cards will have to test it. 
    
    In effect the original code was using i2c_ASN1_INTEGER to get the
    content then converting it to DER. 
    The patch uses i2d_ASN1_INTEGER to get the DER directly. But  the
    error checking is wrong. 
    
    I will fix the error checking.  
    
    For now the #if 1 could be changed to #if OPENSSL_VERSION_NUMBER
    >= 0x10100000L
    but then the code would not be tested until someone built with 1.1.
    The simplification should work with any version of OpenSSL so could
    introduced to the code base once I fix the error checking. 
     
    
    
    On 1/15/2016 3:12 AM, Frank Morgner
      wrote:
    
    
      In src/pkcs15init/pkcs15-oberthur-awp.c:
      > @@ -996,6 +996,20 @@ awp_encode_cert_info(struct sc_pkcs15_card *p15card, struct sc_pkcs15_object *ob
>  	/*
>  	 * serial number
>  	 */
> +#if 1
> +	
> +	/* TODO the der encoding of a ANS1_INTEGER is a TLV, the original code only as using the 
> +	 * i2c_ASN1_INTEGER which is not in OpenSSL 1.1  

      Although I appreciate this simplification, it's not very
        helpful when it is untested. Wouldn't it be safe to simply
        replace i2c_ASN1_INTEGER with i2d_ASN1_INTEGER?
      —
        Reply to this email directly or view
          it on GitHub.
      
        
          
          
        
        
      
    
    
    -- 

 Douglas E. Engert  <DEEngert@gmail.com>